### PR TITLE
Password Reset Email Link

### PR DESCRIPTION
**Pull request by Skibo555**
The endpoint allows users to reset their password by providing the password they used in registration.
​

## Description

The endpoint takes the email the user provided and compare it again the record in the database if it matches any email, if it does, it sends a password reset link into the user's email and returns 200 OK status code. If the email does not exist in the database, it returns "We don't have user with the provided email in our database." and 404 status code.
​

## Related Issue (Link to issue ticket)
https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/51

​

## Motivation and Context

This will allow users to retrieve their email should in case anything happen to their account provided thy know the email address.
​

## How Has This Been Tested?
Yes, I tested it using pytest and Postman.

​

## Screenshots (if appropriate - Postman, etc):

Screenshot 2024-07-24 at 15 10 24
Screenshot 2024-07-24 at 15 15 21
Screenshot 2024-07-24 at 15 15 58
Screenshot 2024-07-24 at 15 53 11

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: api/utils/auth.py
Comment (joboy-dev) [inline api/utils/auth.py:1]: Some of the functions in here is being handled by user.py in services

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Please pull the latest changes from dev branch because some of this code are outdated

**Comment by Skibo555** [comment]
I am currently resolving the conflicts.

File: api/utils/auth.py
Comment (Skibo555) [reply api/utils/auth.py:1]: I am resolving the conflict.

**Pull request closed without merging**